### DISCUSSION TITLE
Usage with home-manager

### DISCUSSION THREAD

**Discussion by Simon-Kenn** [nix-community/kickstart-nix.nvim#46] [Q&A]
Hi,

I leave nixvim for this configuration sample. I love lua for nvim configuration, i did rewrite my entire configuration and during this time i just run `nix run #nvim` in my neovim folder.

But now that i am satisfacte i try to use it as a flake inputs, but i unfortuntly get problem…

After setting the overlays to my flake

I try to use it as a nix-os level:
```nix
environment.systemPackages = with pkgs; [
  nvim-pkg
];
programs.neovim.enable = true;
```

and ultimatly i would like to use it with home-manager, not sure if it's possible

```nix
home.programs.neovim = {
  enable = true;
  package = pkgs.nvim-pkg
};
```

Special note: i use `neovim.url = "git+file:///home/path-to-my-folder-config"; as input, didin't push my repo right now, maybe it cause problem ?

Thanks for this really good sample btw

**Comment by mrcjkb** [comment]
Hey :wave: 

> and ultimatly i would like to use it with home-manager, not sure if it's possible
>
>```nix
>home.programs.neovim = {
>  enable = true;
>  package = pkgs.nvim-pkg
>};
>```

I haven't tried it myself, but don't think that is possible (or you'll at least likely run into issues).
Afaik, home-manager creates a Neovim wrapper, so this would result in it creating a wrapper around `nvim-pkg`, which itself is a Neovim wrapper.

What are you trying to achieve by using it with home-manager?

**Reply by Simon-Kenn** [reply]
I understand for home-manager, i was in the idea to use it for consistency in my global architecture, i like to have my configuration in /home/user/.config

Unfortunatly i did'nt success to use it with nixos, when i start neovim i got a lot of plugin integration not found, start neovim with `nix run ~/fy-flake-neovim-folder` work perfectly,

as i say i add the overlay as flake inputs, add the nvim-pkg to environment system package and try to active it with programs.neovim.enable = true

do you have any idea about what can make them list of module not found ? 

Also at nixos-rebuild where will be the neovim configuration folder ?

Sorry for my really bad english, i hope you undestrand me !

**Reply by mrcjkb** [reply]
> when i start neovim i got a lot of plugin integration not found

This suggests that there's something wrong with the build when you build it in your NixOS config. Usually, it's because you're using a nixpkgs version with an incompatible version of the `wrapNeovimUnstable` function.
(see also the note [in this section](https://github.com/nix-community/kickstart-nix.nvim?tab=readme-ov-file#snowflake-nixos-with-flakes).

> Also at nixos-rebuild where will be the neovim configuration folder ?

The derivation built by kickstart-nix.nvim doesn't have a configuration folder (that wouldn't be reproducible). It all lives in the nix store.

**Comment by Simon-Kenn** [comment]
Sorry was in vaccation

I did fix the inputs.follows, but i have some question if you mind.

This kickstart overlay is designed for install neovim in the system level, right ? 
So is there a way to add some user configuration only, for exemple luasnip template, or is the total configuration intendeed to be the same for every user ?

**Reply by mrcjkb** [reply]
The config is part of the derivation, so it won't pick up an init.lua in your home directory. 
But you can still add plugin scripts, e.g. in `~/.config/nvim/plugin/foo.lua`.
